management_policy.tf

```diff
@@ -0,0 +1,84 @@
+resource "azurerm_storage_management_policy" "this" {
+  count = var.storage_management_policy_rule == null ? 0 : 1
```

Comment:
Doing this as a count is a bad idea. You should make the variable a map of objects and use an arbitrary key.

If you use this method then when you delete a policy then it'll delete and recreate all subsequent policies.